src/pal/src/thread/thread.cpp

```diff
@@ -626,20 +644,28 @@ CorUnix::InternalCreateThread(
     TRACE("default pthread stack size is %d, caller requested %d (default is %d)\n",
           pthreadStackSize, dwStackSize, CPalThread::s_dwDefaultThreadStackSize);
 
-    if (0 == dwStackSize)
-    {
-        dwStackSize = CPalThread::s_dwDefaultThreadStackSize;
-    }
-
 #ifdef PTHREAD_STACK_MIN
     if (PTHREAD_STACK_MIN > pthreadStackSize)
     {
         WARN("default stack size is reported as %d, but PTHREAD_STACK_MIN is "
              "%d\n", pthreadStackSize, PTHREAD_STACK_MIN);
     }
 #endif
-    
-    if (pthreadStackSize < dwStackSize)
+
+    if (dwStackSize == 0)
+    {
+        dwStackSize = CPalThread::s_dwDefaultThreadStackSize;
```

Comment:
I believe it is in case the pthread default is smaller than what CoreCLR intends to use for a default. As the pthread default is configurable by the user and probably appropriate for the system, I suppose it would be better to just use that. I'll remove the hard-coded default.